From f3b744b0b3d565a8706723a71780191388005fdc Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 18 Aug 2018 14:39:34 +0530
Subject: [PATCH 0136/1051] soc: Move SOC entry to under IOMMU

This is a temporary change (not for upstreaming) until
below issue is debugged and fixed properly.

Due to changes in IOMMU infrastructure in 4.14 kernel,
SMMUv3 driver is being probed after PCI devices enumeration
is done. Earlier on 4.9 kernel it used to happen before
ECAM enumeration.

And since in drivers/makefile IOMMU comes after SOC, OcteonTX2
RVU AF driver is being probed before SMMUv3 is initialized
resulting in lots of SMMU errors while kernel is booting.
Though nothing is broken, it looks pretty ackward and raising
false alarms. So moved SOC entry in Makefile to under IOMMU,
so that SMMUv3 driver is initialized first.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/Makefile b/drivers/Makefile
index 24cd47014657..e5f14a6077ed 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -38,9 +38,6 @@ obj-y				+= clk/
 # really early.
 obj-$(CONFIG_DMADEVICES)	+= dma/
 
-# SOC specific infrastructure drivers.
-obj-y				+= soc/
-
 obj-$(CONFIG_VIRTIO)		+= virtio/
 obj-$(CONFIG_XEN)		+= xen/
 
@@ -58,6 +55,9 @@ obj-y				+= char/
 # iommu/ comes before gpu as gpu are using iommu controllers
 obj-$(CONFIG_IOMMU_SUPPORT)	+= iommu/
 
+# SOC specific infrastructure drivers.
+obj-y				+= soc/
+
 # gpu/ comes after char for AGP vs DRM startup and after iommu
 obj-y				+= gpu/
 
-- 
2.17.1

